app/assets/javascripts/discourse/app/lib/uppy-chunked-uploader-plugin.js

```diff
@@ -0,0 +1,215 @@
+
+        resolve(upload);
+      };
+
+      const onChunkComplete = (chunk) => {
+        const cFile = this.uppy.getFile(file.id);
+        if (!cFile) {
+          return;
+        }
+
+        this.uppy.emit("chunk-uploaded", cFile, chunk);
+      };
+
+      const upload = new UppyChunkedUpload(file, {
+        getChunkSize: this.opts.getChunkSize
+          ? this.opts.getChunkSize.bind(this)
+          : null,
+
+        onStart,
+        onProgress,
+        onChunkComplete,
+        onSuccess,
+        onError,
+
+        limit: this.opts.limit || 5,
+        retryDelays: this.opts.retryDelays || [],
+        method: this.method,
+        url: this.url,
+        headers: this.opts.headers,
+      });
+
+      this.uploaders[file.id] = upload;
+      this.uploaderEvents[file.id] = new EventTracker(this.uppy);
+
+      next(() => {
+        if (!file.isPaused) {
+          upload.start();
+        }
+      });
+
+      this._onFileRemove(file.id, (removed) => {
+        this._resetUploaderReferences(file.id, { abort: true });
+        resolve(`upload ${removed.id} was removed`);
+      });
+
+      this._onCancelAll(file.id, () => {
+        this._resetUploaderReferences(file.id, { abort: true });
+        resolve(`upload ${file.id} was canceled`);
+      });
+
+      this._onFilePause(file.id, (isPaused) => {
+        if (isPaused) {
+          upload.pause();
+        } else {
+          next(() => {
+            upload.start();
+          });
+        }
+      });
+
+      this._onPauseAll(file.id, () => {
+        upload.pause();
+      });
+
+      this._onResumeAll(file.id, () => {
+        if (file.error) {
+          upload.abort();
+        }
+        next(() => {
+          upload.start();
+        });
+      });
+
+      // Don't double-emit upload-started for restored files that were already started
+      if (!file.progress.uploadStarted || !file.isRestored) {
+        this.uppy.emit("upload-started", file);
+      }
+    });
+  }
+
+  _onFileRemove(fileID, cb) {
+    this.uploaderEvents[fileID].on("file-removed", (file) => {
+      if (fileID === file.id) {
+        cb(file.id);
+      }
+    });
+  }
+
+  _onFilePause(fileID, cb) {
+    this.uploaderEvents[fileID].on("upload-pause", (targetFileID, isPaused) => {
+      if (fileID === targetFileID) {
+        cb(isPaused);
+      }
+    });
+  }
+
+  _onPauseAll(fileID, cb) {
+    this.uploaderEvents[fileID].on("pause-all", () => {
+      if (!this.uppy.getFile(fileID)) {
+        return;
+      }
+      cb();
+    });
+  }
+
+  _onCancelAll(fileID, cb) {
+    this.uploaderEvents[fileID].on("cancel-all", () => {
+      if (!this.uppy.getFile(fileID)) {
+        return;
+      }
+      cb();
+    });
+  }
+
+  _onResumeAll(fileID, cb) {
+    this.uploaderEvents[fileID].on("resume-all", () => {
+      if (!this.uppy.getFile(fileID)) {
+        return;
+      }
+      cb();
+    });
+  }
+
+  _upload(fileIDs) {
+    if (fileIDs.length === 0) {
```

Comment:
This condition seems unnecessary. If `fileIds` is empty the map will return an empty array, and `Promise.all([])` will resolve.